MoreLinq.NoConflictGenerator/Program.cs

```diff
@@ -0,0 +1,225 @@
+                                .WithExpressionBody(
+                                    ArrowExpressionClause(
+                                        InvocationExpression(
+                                            MemberAccessExpression(
+                                                SyntaxKind.SimpleMemberAccessExpression,
+                                                IdentifierName("MoreEnumerable"),
+                                                IdentifierName(md.Identifier)),
+                                            ArgumentList(SeparatedList(
+                                                from p in md.ParameterList.Parameters
+                                                select Argument(IdentifierName(p.Identifier)))))
+                                            .WithLeadingTrivia(Space))
+                                        .WithLeadingTrivia(Whitespace(indent3)))
+                                .WithSemicolonToken(ParseToken(";").WithTrailingTrivia(LineFeed))
+                }
+                into m
+                select (!noClassLead ? $@"
+    /// <summary><c>{m.Name}</c> extension.</summary>
+
+    [GeneratedCode(""{thisAssemblyName.Name}"", ""{thisAssemblyName.Version}"")]" : null) + $@"
+    public static partial class {m.Name}Extension
+    {{
+{string.Join(null, from mo in m.Overloads select mo.ToFullString())}
+    }}";
+
+            var template = $@"
+#region License and Terms
+// MoreLINQ - Extensions to LINQ to Objects
+// Copyright (C) 2008 Jonathan Skeet.
+// Portions Copyright (C) 2009 Atif Aziz, Chris Ammerman, Konrad Rudolph.
+// Portions Copyright (C) 2010 Johannes Rudolph, Leopold Bushkin.
+// Portions Copyright (C) 2015 Felipe Sateler, ""sholland"".
+// Portions Copyright (C) 2016 Leandro F. Vieira (leandromoh).
+// Portions Copyright (C) Microsoft. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the ""License"");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an ""AS IS"" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+// This code was generated by a tool. Any changes made manually will be lost
+// the next time this code is regenerated.
+
+namespace MoreLinq.NoConflict
+{{
+{string.Join("\n", imports)}
+{string.Join("\n", classes)}
+}}
+            ";
+
+            Console.WriteLine(template.Trim()
+                                      // normalize line endings
+                                      .Replace("\r", string.Empty)
+                                      .Replace("\n", Environment.NewLine));
```

Comment:
We have [`.gitattributes` with line-ending normalisation configured](https://github.com/morelinq/MoreLINQ/blob/6ee347f635cd48429fe93ba89be40941dbdd11a4/.gitattributes#L2) so it should be harmless. You you can see in the [case of a failing build](https://ci.appveyor.com/project/raboof/morelinq/build/525#L245):

```
warning: CRLF will be replaced by LF in MoreLinq/NoConflict.ToDataTable.g.cs.
The file will have its original line endings in your working directory.
warning: CRLF will be replaced by LF in MoreLinq/NoConflict.g.cs.
The file will have its original line endings in your working directory.
```

The build failed (which I am still trying to understand why) because it detected generated code that's uncommitted. Strangely, the same commit succeeded in a separate build. 🤔

Reply:
Seems [the build failed in one case](https://ci.appveyor.com/project/raboof/morelinq/build/525#L245) because `master` has changes unrelated to this PR that caused the generated code output to change when this PR is merged with `master`. While a merge of `master` with this PR will help here, it may not be realistic for other PRs and may give false negatives. 😒